python/ray/data/datasource/file_datasink.py

```diff
@@ -79,6 +79,9 @@ def open_output_stream(self, path: str) -> "pyarrow.NativeFile":
         return self.filesystem.open_output_stream(path, **self.open_stream_args)
 
     def on_write_start(self) -> None:
+        self.has_created_dir = self._create_dir(self.path)
+
+    def _create_dir(self, dest) -> bool:
```

Comment:
Do we use the `bool` return value anywhere? If not, should this just be `None`?
```suggestion
    def _create_dir(self, dest) -> None:
```